File: ko_chat/backend/src/main/kotlin/com/kochat/global/application/messaging/dto/MessagingOperationsSnapshot.kt
package com.kochat.global.application.messaging.dto

import java.time.LocalDateTime

data class MessagingOperationsSnapshot(
    val outbox: OutboxStatusSnapshot,
    val processedEvents: ProcessedEventSnapshot,
    val dlq: DlqStatusSnapshot,
    val consumerLag: List<ConsumerLagSnapshot>,
    val lagHealthy: Boolean,
    val checkedAt: LocalDateTime,
)

File: ko_chat/frontend/src/types/messaging/messaging-operations-snapshot.ts
import type { OutboxStatusSnapshot, ProcessedEventSnapshot, DlqStatusSnapshot } from './messaging-snapshots'
import type { ConsumerLagSnapshot } from './consumer-lag-snapshot'

export interface MessagingOperationsSnapshot {
  kafkaEnabled: boolean
  outbox: OutboxStatusSnapshot
  processedEvents: ProcessedEventSnapshot
  dlq: DlqStatusSnapshot
  consumerLag: ConsumerLagSnapshot[]
  lagHealthy: boolean
  checkedAt: string
}
